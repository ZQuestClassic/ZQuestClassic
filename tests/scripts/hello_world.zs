global script Global
{
    void run()
    {
        int number = 123;
        printf("hello world: %d\n", number);
    }
}
