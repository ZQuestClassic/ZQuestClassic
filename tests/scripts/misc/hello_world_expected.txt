stderr:

Compiling 'hello_world.zs'
Pass 1: Parsing
Pass 2: Preprocessing
Pass 3: Registration
Pass 4: Analyzing Code
Pass 5: Checking code paths
Pass 6: Generating object code
Pass 7: Assembling
Success!
Compile finished with exit code '0' (success)

stdout:

{
  "success": true,
  "diagnostics": [],
  "metadata": {
    "currentFileSymbols": 3,
    "symbols": 4,
    "identifiers": 5,
    "elided": true
  }
}

zasm:

QUIT; void run() Body
PUSHARGSV 0 0.0002; void run() Body Start
SETR D4 SP2
STOREV 123 0
ALLOCATEMEMV D2 17 0; InlineFunc[void printf(char32[])] Params Start
WRITEPODSTRING D2 "hello world: %d\n"
PUSHR D2
LOAD D2 0
PUSHVARGR D2; InlineFunc[void printf(char32[])] Params End
PRINTFVARG; InlineFunc[void printf(char32[])] Body Start
POP D5; InlineFunc[void printf(char32[])] Body End
POPARGS D5 0.0002
QUIT; void run() Body End
